clang/include/clang/Basic/AttrDocs.td

```diff
@@ -4277,6 +4281,46 @@ See Sub-groups for NDRange Parallelism proposal in sycl/doc/extensions/sub_group
   }];
 }
 
+def IntelNamedSubGroupSizeDocs : Documentation {
+  let Category = DocCatFunction;
+  let Content = [{
+The ``[[intel::named_sub_group_size(NAME)]]`` attribute works with the
+``[[intel::sub_group_size(N)]]`` attribute to define the specific subgroup size
+for the kernel. The ``[[intel::named_sub_group_size(NAME)]]`` form accepts a
+required parameter of either ``auto`` or ``primary``.
+
+``auto`` specifies that the implementation is free to select any of the valid
+sub-group sizes associated with the device to which the kernel is submitted.
+``primary`` specifies that the implementation should select the device's primary
+sub-group size as reported by ``info::device::primary_sub_group_size``.
+
+This attribute may not be combined with ``[[intel::sub_group_size(N)]]``, as
```

Comment:
The first sentence of the docs make it sound like these two go hand-in-hand. I'll suggest a change there to see if that helps.

Reply:
Cool, thanks!